bot/common/threads/onboarding.py

```diff
@@ -98,56 +118,315 @@ async def handle_emoji(self, raw_reaction):
         return _handle_skip_emoji(raw_reaction, self.guild_id)
 
 
-class AddUserTwitterStep(BaseStep):
-    """Step to submit twitter name for the govrn profile"""
+class AddUserAccountStep(BaseStep):
+    """Step which prompts user for an account name. Derived classes"""
 
-    name = StepKeys.ADD_USER_TWITTER.value
+    """override the message and cache name for the response"""
 
-    def __init__(self, guild_id):
+    def __init__(self, user_id, guild_id, cache):
         super().__init__()
+        self.user_id = user_id
         self.guild_id = guild_id
+        self.cache = cache
 
     async def send(self, message, user_id):
         channel = message.channel
         sent_message = await channel.send(
-            "What twitter handle would you like to associate with this guild!"
+            "What %s would you like to associate with this guild?"
+            % self.get_account_descriptor()
```

Comment:
Styling tip prefer fstrings over the classic string interpolation method

Reply:
done